pkg/connector/project_role.go

```diff
@@ -0,0 +1,263 @@
+		return nil, "", nil, wrapError(err, "failed to parse project role ID")
+	}
+
+	project, err := u.client.GetProject(ctx, projectID)
+	if err != nil {
+		return nil, "", nil, wrapError(err, "failed to get project")
+	}
+
+	role, err := u.client.GetRole(ctx, roleID)
+	if err != nil {
+		return nil, "", nil, wrapError(err, "failed to get role")
+	}
+
+	assigmentOptions := []ent.EntitlementOption{
+		ent.WithGrantableTo(resourceTypeUser, resourceTypeGroup),
+		ent.WithDescription(fmt.Sprintf("Assigned to %s role on the %s project", role.Name, project.Name)),
+		ent.WithDisplayName(fmt.Sprintf("%s Assignment", resource.DisplayName)),
+	}
+	rv = append(rv, ent.NewAssignmentEntitlement(resource, assignedEntitlement, assigmentOptions...))
+
+	return rv, "", nil, nil
+}
+
+func (p *projectRoleResourceType) Grants(ctx context.Context, resource *v2.Resource, pt *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	projectID, roleID, err := parseProjectRoleID(resource.Id.Resource)
+	if err != nil {
+		return nil, "", nil, wrapError(err, "failed to parse project role ID")
+	}
+
+	var rv []*v2.Grant
+
+	projectRoleActors, _, err := p.client.Jira().Role.GetRoleActorsForProject(ctx, projectID, roleID)
+	if err != nil {
+		return nil, "", nil, wrapError(err, "failed to get role actors for project")
+	}
+
+	for _, actor := range projectRoleActors {
+		var g *v2.Grant
+		switch actor.Type {
+		case atlassianUserRoleActor:
+			userActor := &v2.ResourceId{
+				ResourceType: resourceTypeUser.Id,
+				Resource:     actor.ActorUser.AccountID,
+			}
+			g = grant.NewGrant(resource, assignedEntitlement, userActor)
+
+		case atlassianGroupRoleActor:
+			groupActor := &v2.ResourceId{
+				ResourceType: resourceTypeGroup.Id,
+				Resource:     actor.ActorGroup.GroupID,
+			}
+			g = grant.NewGrant(resource, assignedEntitlement, groupActor, grant.WithAnnotation(&v2.GrantExpandable{
+				EntitlementIds:  []string{fmt.Sprintf("group:%s:%s", actor.ActorGroup.GroupID, memberEntitlement)},
+				ResourceTypeIds: []string{resourceTypeUser.Id},
+			}))
+
+		default:
+			l.Warn("unknown role actor type", zap.String("type", actor.Type))
+			continue
+		}
+
+		rv = append(rv, g)
+	}
+
+	return rv, "", nil, nil
+}
+
+func (p *projectRoleResourceType) List(ctx context.Context, _ *v2.ResourceId, token *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	bag, offset, err := parsePageToken(token.Token, &v2.ResourceId{ResourceType: resourceTypeProjectRole.Id})
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	projects, _, err := p.client.Jira().Project.Find(ctx, jira.WithStartAt(int(offset)), jira.WithMaxResults(resourcePageSize))
+	if err != nil {
+		return nil, "", nil, wrapError(err, "failed to get projects")
+	}
+
+	var ret []*v2.Resource
+	for _, prj := range projects {
+		project, err := p.client.GetProject(ctx, prj.ID)
+		if err != nil {
+			return nil, "", nil, wrapError(err, "failed to get project")
```

Comment:
Nit: please include the project id in this error. We should be able to get the project if it was returned in a find but it hasn't always been the case. I know this is mostly moved code.